src/test/js/karaoke-noStages.js

```diff
@@ -26,15 +26,19 @@ module.exports = {
         browser.waitForElementVisible('code')
             .click('code')
             .keys(browser.Keys.UP_ARROW)
-            .getText('code', function (result) {
-                const text = result.value;
-                this.pause(1000)
-                    .waitForElementVisible('code')
-                    .getText('code', function (result) {
-                        this.assert.equal(text, result.value)
-                    });
-
-            });
+            .elements('css selector', 'div.result-item', function (resutlItems) {
+                var results = resutlItems.value.length;
+                // to validate that we left follow, give it some time and then count the elements again
+                this.pause(3000)
+                    .elements('css selector', 'code', function (codeCollection) {
+                        // JENKINS-36700 there can only be one code view open in follow stopped
+                        this.assert.equal(codeCollection.value.length, 1);
+                    })
+                    .elements('css selector', 'div.result-item', function (resutlItemsCompare) {
+                        // there should not be more items then we had before
+                        this.assert.equal( resutlItemsCompare.value.length, results);
+                    })
+                });
```

Comment:
Is any of this reusable ?

Those `pause`s are dodgy imo ... resulting in flaky tests. Is there no other way?

Did you try using `waitForElementVisible` etc? The have a 20s (iirc) timeout, but will return as soon as the element is visible.

The `pause` can either be way too much (waiting 3s when it really only needs to wait 30ms), or way too little (something random happened to cause a longer delay).